client/my-sites/plans-features-main/hooks/data-store/use-pricing-meta-for-grid-plans.ts

```diff
@@ -119,13 +112,13 @@ const usePricingMetaForGridPlans: UsePricingMetaForGridPlans = ( {
 					 */
 					if ( purchasedPlan ) {
 						const isMonthly = purchasedPlan.billPeriodDays === PLAN_MONTHLY_PERIOD;
-						const isYearly = purchasedPlan.billPeriodDays === PLAN_ANNUAL_PERIOD;
 
 						if ( isMonthly && monthlyPrice !== purchasedPlan.priceInteger ) {
 							monthlyPrice = purchasedPlan.priceInteger;
-							yearlyPrice = calculateYearlyPrice( TERM_ANNUALLY, purchasedPlan.priceInteger );
-						} else if ( isYearly && yearlyPrice !== purchasedPlan.priceInteger ) {
-							monthlyPrice = calculateMonthlyPrice( TERM_ANNUALLY, purchasedPlan.priceInteger );
+							yearlyPrice = parseFloat( ( purchasedPlan.priceInteger * 12 ).toFixed( 2 ) );
+						} else if ( yearlyPrice !== purchasedPlan.priceInteger ) {
+							const months = Math.round( purchasedPlan.billPeriodDays / PLAN_MONTHLY_PERIOD );
+							monthlyPrice = parseFloat( ( purchasedPlan.priceInteger / months ).toFixed( 2 ) );
```

Comment:
Removed the previous `calculateYearlyPrice`.
Also removed the `calculateMonthlyPrice` which relies on there being a term property, which we don't have on the purchasedPlan object. We do have `billPeriodDays` so we can calculate the monthly price manually.

Reply:
Unfortunately that doesn't work because the rounding isn't always correct.  Here's what I got for a three-year plan:

![three-year-plan](https://github.com/Automattic/wp-calypso/assets/235183/c02e6ed7-3159-4077-802c-41d8635c40f3)

It should be $8.25, but it winds up displaying $8.49 because the code is using 35 months rather than 36.

I think it might be necessary to convert to the term constant somehow after all?

Reply:
> I think it might be necessary to convert to the term constant somehow after all?

What would you say about rounding up instead?

Reply:
I guess rounding up will work through a 3-year plan, but eventually won't work at some point beyond that.  (Won't work for a 100-year plan, for example!  Although I'm not sure if we ever display the price for that plan in this way.)

I think to make it cleaner and more future-proof, it might be better to just define a function that works kind of in the exact opposite way as the one below, and then use it to go from the `billPeriodDays` to the term?

https://github.com/Automattic/wp-calypso/blob/3bbf2be746bbf3c91ac7266a0594277a73ebc5d7/packages/calypso-products/src/plans-utilities.ts#L42-L69

Reply:
Sounds good! Thanks!
I added a `getTermFromDuration` and used it together with `getBillingMonthsForTerm` to get the months from the available constants.